src/tools/rbd_nbd/win32/rbd-nbd.cc

```diff
@@ -0,0 +1,1038 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```

Comment:
For the Linux `rbd-nbd` daemon, we are planning to have a single daemon handle all the RBD NBD devices on the node. It seems like this `rbd-nbd` service should follow a similar model from the start. i.e. `rbd-nbd map/unmap` should really just send a notification message to an already running `rbd-nbd service` via a control message. Or alternatively, `rbd-nbd` on Windows is just the service daemon (no CLI inputs) and `rbd device map/unmap` for NBD calls some helper methods to send the control messages.

Reply:
Thanks for the heads up. To be honest, we'd rather keep it like this for the initial implementation so that we don't delay those PRs any further. We'd like to focus on performance for the next couple of weeks. Our goal is to outperform the iSCSI gateway. ``ceph-dokan`` already provides good performance, significantly better than ``ceph-fuse``.

Here are some benchmark results: http://paste.openstack.org/raw/792539/

Reply:
My fear is that once established, "rbd-nbd map/unmap/list-mapped" will stick around forever and I really don't want to see that occur. I wasn't suggesting to use the Ceph's interprocess communication framework -- you are already using some other Windows-centric IPC it seems like.

Reply:
FWIW, we're still using unix sockets for the admin socket interface on Windows Server 2019.

Reply:
The latest iteration no longer uses NBD. We're still using one process per mapping though, for the time being.

Reply:
"rbd device map" works with the new "rbd-wnbd" tool, so I think I can mark this as resolved.